Community/Tdarr_Plugin_advo_FFMPEG_NVENC_Tiered_MKV_with_rollback.js

```diff
@@ -0,0 +1,563 @@
+        file.ffProbeData.streams[i].codec_name.toLowerCase() === 'truehd'
+        || (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'dts'
+          && file.ffProbeData.streams[i].profile.toLowerCase() === 'dts-hd ma')
+        || (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'aac'
+          && file.ffProbeData.streams[i].sample_rate.toLowerCase() === '44100'
+          && file.ffProbeData.streams[i].codec_type.toLowerCase() === 'audio')
+      ) {
+        maxmux = ' -max_muxing_queue_size 9999';
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+    }
+
+    // mitigate errors due to embeded pictures
+    try {
+      if (file.ffProbeData.streams[i].codec_type.toLowerCase() === 'video'
+        && (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'png'
+          || file.ffProbeData.streams[i].codec_name.toLowerCase() === 'bmp'
+          || file.ffProbeData.streams[i].codec_name.toLowerCase() === 'mjpeg')
+      ) {
+        map += ` -map -0:${i} `;
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+    }
+  }
+
+  if (file.video_resolution === '480p' || file.video_resolution === '576p') {
+    cqv = inputs.sdCQV;
+    maxBitrate = inputs.sdMaxBitrate;
+    transcode = 1;
+  }
+
+  if (file.video_resolution === '720p') {
+    cqv = inputs.hdCQV;
+    maxBitrate = inputs.hdMaxBitrate;
+    transcode = 1;
+  }
+
+  if (file.video_resolution === '1080p') {
+    cqv = inputs.fullhdCQV;
+    maxBitrate = inputs.fullhdMaxBitrate;
+    transcode = 1;
+  }
+
+  if (file.video_resolution === '4KUHD') {
+    cqv = inputs.uhdCQV;
+    maxBitrate = inputs.uhdMaxBitrate;
+    transcode = 1;
+  }
+
+  // need to find a reliable test for HDR content
+  if (videoStream.color_transfer !== undefined
+    && (videoStream.color_transfer.toLowerCase() === 'smpte2084'
+    || videoStream.color_transfer.toLowerCase() === 'arib-std-b67')) {
+    isHDR = true;
+    if (filterCommand !== '') {
+      filterCommand += ',';
+    }
+    filterCommand += 'zscale=t=linear:npl=100,format=gbrpf32le,zscale=p=bt709,'
+      + 'tonemap=tonemap=hable:desat=0,zscale=t=bt709:m=bt709:r=tv,format=p010le';
+  }
+
+  // check if the file is eligible for transcoding
+  // if true the neccessary response values will be changed
+  if (transcode === 1) {
+    if (filterCommand !== '') {
+      response.preset = ''; // hardware decoding doesn't seem to work with the HDR tonemapping filter. Not sure why
+      filterCommand = `-vf "${filterCommand}"`;
+    }
+
+    response.preset += `<io> ${map} -dn -c:v hevc_nvenc -profile:v main10 -preset p5 -tune hq -pix_fmt p010le `
+      + `${filterCommand} -rc:v vbr `
+      + `-multipass 2 -bufsize 600M -b:v 0 -maxrate:v ${maxBitrate} -qmin 0 -qmax ${cqv} -cq:v ${cqv} `
+      + `-rc-lookahead 32 -nonref_p 1 -a53cc 0 -threads 0 ${extraOptions} ${subcli} ${maxmux} `
+      + '-metadata ADVOVTRANSCODEDONE=true';
```

Comment:
Might want to add aq-mode 3, this helps a ton in dark scenes.

Reply:
I looked into this; a couple of things:
1. nvenc doesn't support aq-mode 3 (it just does nothing)
2. I tried using aq-mode 2, but the file size was exactly the same as without. I also couldn't see any visible difference.
3. I'm doing 10 bit encoding, which already does quite well in dark scenes.